db/generate_data.py

```
@@ -0,0 +1,732 @@
+    cursor = conn.cursor()
+    
+    # Additional indexes
+    indexes = [
+        "CREATE INDEX IF NOT EXISTS idx_sales_date_status ON sales(DATE(created_at), sale_status_desc)",
+        "CREATE INDEX IF NOT EXISTS idx_product_sales_product_sale ON product_sales(product_id, sale_id)",
+    ]
+    
+    for idx in indexes:
+        try:
+            cursor.execute(idx)
+        except:
```

Ruff (0.14.2)

Do not use bare `except`

(E722)